slotfills/sections/richtext-field.jsx

```diff
@@ -0,0 +1,65 @@
+/* global tinyMCEPreInit */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { PanelRow } from '@wordpress/components';
+import { Editor } from '@tinymce/tinymce-react';
+
+const RichtextField = ({
+  field: {
+    name,
+  },
+  valueHook,
+  index = null,
+  label = '',
+}) => {
+  const [value, setValue] = index !== null ? valueHook(index) : valueHook(name);
+  const initialvalue = typeof value === 'object' ? value[name] : value;
+  const [stateValue, setStateValue] = useState(initialvalue);
+
+  const updateValue = () => { // eslint-disable-line no-unused-vars
+    const newValue = typeof value === 'object' ? { [name]: stateValue } : stateValue;
+    setValue(newValue);
+  };
+
+  return (
+    <PanelRow>
+      <div style={{ flexDirection: 'column' }}>
+        <label htmlFor={`${name}_${index}`}>
+          {label}
+        </label>
+        <Editor
+          value={stateValue}
+          tinymceScriptSrc={`${tinyMCEPreInit.baseURL}/tinymce.min.js`}
+          init={{
+            height: 200,
+            menubar: false,
+            plugins: [
+              'lists link image fullscreen media paste',
+            ],
+            toolbar: 'formatselect bold italic bullist numlist blockquote alignleft aligncenter alignright link fullscreen',
+            content_style: 'body { font-family:Helvetica,Arial,sans-serif; font-size:14px }',
+          }}
+          onEditorChange={setStateValue}
```

Comment:
The docs say that another way to do this is to use a ref, and that using `onEditorChange` can have performance problems because the value needs to be cast to a string on every keystroke, saved to state, then unpacked from a string and re-flowed. Is using the TinyMCE editor as a controlled component required, or can we use a ref instead?

Reply:
I couldn't get the ref to work and have constant updating of the value, so I left it as a controlled component.